File: Arrays-And-Strings/Challenge-9.js
//0(N) time complexity and 0(N) space complexity

const isSubString = (string1, string2) => {
  if (string1.length !== string2.length) {
    return false;
  }
  let subStringChecker = `${string1}${string1}`;
  return subStringChecker.includes(string2);
};

console.log(isSubString('Waterbottle', 'erbottleWat')); // true
console.log(isSubString('abc', 'cba')); // false

File: Arrays-And-Strings/Challenge-9.py
# 0(N) time complexity and 0(N) space complexity

def is_substring(string1,string2):
  if len(string1) != len(string2):
    return False

  sub_string_checker = string1 + string1
  return string2 in sub_string_checker

print(is_substring('Waterbottle', 'erbottleWat')) # True
print(is_substring('abc', 'cba')) # False
